### PR TITLE
Refactored with hooks

### PR DESCRIPTION
**Pull request by ShoiraTa**
# Refactored
Refactored the Calculator.js from a class based component to a functional component with hooks.

### CODE REVIEW

**Comment by ShoiraTa** [comment]
Dear reviewer, in some reviewers machine `npm start` gives an error, in this case please run `npm audit fix --force`

File: src/componets/CalculatorUI.js [hunk 1]
```diff
@@ -1,59 +1,49 @@
 /* eslint-disable react/prefer-stateless-function */
 /* eslint-disable no-useless-constructor */
```
Comment (D4nielJ) [inline src/componets/CalculatorUI.js:2]: [OPTIONAL] I think disabling these rules is not necessary anymore. Let's remove them so our code remains clean.

File: src/componets/CalculatorUI.js [hunk 2]
```diff
@@ -1,59 +1,49 @@
...
+          <button type="button" onClick={handleClick} className="bg-gray single-cell">5</button>
+          <button type="button" onClick={handleClick} className="bg-gray single-cell">6</button>
+          <button type="button" onClick={handleClick} className="bg-orange single-cell">-</button>
+          <button type="button" onClick={handleClick} className="bg-gray single-cell">1</button>
+          <button type="button" onClick={handleClick} className="bg-gray single-cell">2</button>
+          <button type="button" onClick={handleClick} className="bg-gray single-cell">3</button>
+          <button type="button" onClick={handleClick} className="bg-orange single-cell">+</button>
+          <button type="button" onClick={handleClick} className="bg-gray zero single-cell">0</button>
+          <button type="button" onClick={handleClick} className="bg-gray single-cell">.</button>
+          <button type="button" onClick={handleClick} className="bg-orange single-cell">=</button>
```
Comment (D4nielJ) [inline src/componets/CalculatorUI.js:42]: [OPTIONAL] I think you could create a different button component and an array with the different buttons. Then map through the array to print all of the different buttons. This way you can practice how to pass props to children and how to dynamically render components.

File: src/componets/CalculatorUI.js (hunk 1 above)
Comment (D4nielJ) [reply src/componets/CalculatorUI.js:2]: [OPTIONAL] I recommend you not to remove `React.StrictMode` from index.js. This component is very helpful when developing apps as it encourages you to follow best practices and warns you of potential problems with your app. [You can read more about here.](https://reactjs.org/docs/strict-mode.html)

Comment (D4nielJ) [reply src/componets/CalculatorUI.js:2]: [OPTIONAL] When using the modulo with 0 it crashes the app. I think this should work in the same manner as dividing by 0 does. This recommendation does encourage you to modify `operate.js`, which was given to you, so feel free not to follow it.

image

**Review by D4nielJ — APPROVED** [review]
## Hello @ShoiraTa, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**